lang/syn/src/parser/spl_interface.rs

```diff
@@ -0,0 +1,59 @@
+#[cfg(feature = "interface-instructions")]
+use syn::{Meta, NestedMeta, Path};
+
+#[cfg(not(feature = "interface-instructions"))]
+pub fn parse(_attrs: &[syn::Attribute]) -> Option<[u8; 8]> {
+    None
+}
+
+#[cfg(feature = "interface-instructions")]
+pub fn parse(attrs: &[syn::Attribute]) -> Option<[u8; 8]> {
+    let interfaces: Vec<[u8; 8]> = attrs
+        .iter()
+        .filter_map(|attr| {
+            if let Ok(Meta::List(meta_list)) = attr.parse_meta() {
+                if meta_list.path.is_ident("interface") {
+                    if let Some(NestedMeta::Meta(Meta::Path(path))) = meta_list.nested.first() {
+                        return Some(parse_interface_instruction(path));
+                    }
+                }
+            }
+            None
+        })
+        .collect();
+    if interfaces.len() > 1 {
+        panic!("An instruction can only implement one interface instruction");
+    } else if interfaces.is_empty() {
+        None
```

Comment:
Should `#[interface()]` be allowed?

Reply:
My opinion is no. I think as of right now, if you use `#[interface(..)]`, you should _only_ be able to add one of the SPL Transfer Hook interface discriminators, that's it.

If we decide to add support for more interfaces in the future, or expand on this attribute, that's fine, but it should occur outside the scope of this change.

With that being said, I think this should be caught with an error, and right now it might not be. I'll add this.

Reply:
Added check!